ttnn/cpp/ttnn/operations/data_movement/permute/permute.cpp

```diff
@@ -193,23 +175,27 @@ ttnn::Tensor ExecutePermute::invoke(
     const ttnn::Tensor& input_tensor,
     tt::stl::Span<const int64_t> dims,
     const std::optional<MemoryConfig>& memory_config,
-    bool composite,
     const std::optional<float>& pad_value) {
-    if (composite) {
-        return detail::composite_invoke(input_tensor, dims, memory_config, pad_value);
-    }
-
-    const bool initial_input_tensor_on_device = detail::is_on_device(input_tensor);
-    const auto input_layout = input_tensor.get_layout();
     const auto input_rank = input_tensor.get_logical_shape().rank();
-
     TT_FATAL(
         input_rank == dims.size(),
         "The number of dimensions in the tensor input does not match the length of the desired ordering");
+    TT_FATAL(detail::is_on_device(input_tensor), "Tensor must already be on device");
+
+    SmallVector<uint32_t> normalized_dims(dims.size());
+    std::transform(dims.begin(), dims.end(), normalized_dims.begin(), [input_tensor](std::int64_t idx) {
+        return input_tensor.get_logical_shape().get_normalized_index(idx);
+    });
+    if (detail::is_permute_nop(input_tensor, normalized_dims)) {
+        return ttnn::to_memory_config(input_tensor, memory_config.value_or(input_tensor.memory_config()));
+    }
+
+    auto padded_shape = input_tensor.get_padded_shape();
```

Comment:
does not seem to be used anywhere, maybe can be removed

Reply:
Done